contracts/lazyMint/LazyMintERC1155.sol

```diff
@@ -0,0 +1,606 @@
+    function uri(uint256 _tokenId) public view override returns (string memory _tokenURI) {
+
+        for(uint256 i = 0; i < baseURIIndices.length; i += 1) {
+            if(_tokenId < baseURIIndices[i]) {
+                return string(abi.encodePacked(baseURI[baseURIIndices[i]], _tokenId.toString()));
+            }
+        }
+        
+        return "";
+    }
+
+    ///     =====   External functions  =====
+
+    /**
+     *  @dev Lets an account with `MINTER_ROLE` mint tokens of ID from `nextTokenIdToMint` 
+     *       to `nextTokenIdToMint + _amount - 1`. The URIs for these tokenIds is baseURI + `${tokenId}`.
+     */
+    function lazyMint(uint256 _amount, string calldata _baseURIForTokens) external onlyMinter {
+        uint256 startId = nextTokenIdToMint;
+        uint256 baseURIIndex = startId + _amount;
+
+        nextTokenIdToMint = baseURIIndex;
+        baseURI[baseURIIndex] = _baseURIForTokens;
+        baseURIIndices.push(baseURIIndex);
+        
+        emit LazyMintedTokens(startId , startId + _amount - 1, _baseURIForTokens);
+    }
+    
+    /// @dev Lets an account claim a given quantity of tokens, of a single tokenId.
+    function claim(
+        uint256 _tokenId,
+        uint256 _quantity,
+        bytes32[] calldata _proofs
+    ) 
+        external
+        payable
+        nonReentrant 
+        whenNotPaused
+    {
+        // Get the claim conditions.
+        uint256 activeConditionIndex = getIndexOfActiveCondition(_tokenId);
+        MintCondition memory mintCondition = mintConditions[_tokenId].mintConditionAtIndex[activeConditionIndex];
+
+        // Verify claim validity. If not valid, revert.
+        verifyClaimIsValid(
+            _tokenId,
+            _quantity,
+            _proofs,
+            activeConditionIndex,
+            mintCondition
+        );
+
+        // If there's a price, collect price.        
+        collectClaimPrice(mintCondition, _quantity, _tokenId);
+
+        // Mint the relevant tokens to claimer.
+        transferClaimedTokens(mintCondition, activeConditionIndex, _tokenId, _quantity);
+        
+        emit ClaimedTokens(activeConditionIndex, _tokenId, _msgSender(), _quantity);
+    }
+
+    /// @dev Lets a module admin set mint conditions for a given tokenId.
+    function setPublicMintConditions(
+        uint256 _tokenId, 
+        MintCondition[] calldata _conditions
+    ) 
+        external
+        onlyModuleAdmin 
+    {
+        // make sure the conditions are sorted in ascending order
+        uint256 lastConditionStartTimestamp = 0;
+        uint256 indexForCondition = mintConditions[_tokenId].nextConditionIndex;
+
+        for (uint256 i = 0; i < _conditions.length; i++) {
+            
+            require(
+                lastConditionStartTimestamp == 0 
+                    || lastConditionStartTimestamp < _conditions[i].startTimestamp,
+                "LazyMintERC1155: startTimestamp must be in ascending order."
```

Comment:
I'll remove the contract name from the require statements for now; it'll help size down the contract.

In general, I disagree with the comment. I think it's a good practice to name the contract in the require message to immediately tell a developer where in the inheritance hierarchy the error occurs (e.g. If an error occurs in an inherited contract e.g. `ERC1155`, error messages name the `ERC1155` contract).

Reply:
See 81919fbf5c57abbd65f1ce875645e80ce95fa6d3

Reply:
the way i see it: it's only useful and would only adds context if it's an abstract class. exactly how you put it. for example, for openzeppelin; we extend it, so that makes sense. it quickly help us navigate through all the classes that we extend.
in our case, our contract is a top level contract, so it should be obvious already.

however, i do agree that it creates inconsistency pattern for error messages. trade off between consistency vs contract size :)